tests/phpunit/tests/strings-in-block/test-html.php

```diff
@@ -409,6 +409,43 @@ public function it_should_update_innerContent_containing_a_slash() {
 		$this->assertEquals( '<div>' . $translated_block_inner_HTML . '</div>', $updated_block->innerHTML );
 	}
 
+	/**
+	 * @test
+	 * @group wpmlcore-6643
+	 */
+	public function it_should_update_and_NOT_escape_invalid_href() {
+		$block_name = 'core/some_block';
+
+		$config_option = \Mockery::mock( 'WPML_Gutenberg_Config_Option' );
+		$config_option->shouldReceive( 'get' )
+		              ->andReturn( [ $block_name => [ 'xpath' => [ '//div' ] ] ] );
+
+		$strings_in_block = new HTML( $config_option );
+
+		$target_lang                 = 'de';
+		$original_block_inner_HTML   = '<a href="[some-shortcode]">Click me</a>';
+		$translated_block_inner_HTML = '<a href="[some-shortcode]">Click me translated</a>';
+
+
+		$strings = array(
+			md5( $block_name . $original_block_inner_HTML ) => array(
+				$target_lang => array(
+					'value'  => $translated_block_inner_HTML,
+					'status' => (string) ICL_TM_COMPLETE,
+				)
+			)
+		);
+
+		$block               = \Mockery::mock( 'WP_Block_Parser_Block' );
+		$block->blockName    = $block_name;
+		$block->innerHTML    = '<div>' . $original_block_inner_HTML . '</div>';
+		$block->innerContent = [ $block->innerHTML ];
+
+		$updated_block = $strings_in_block->update( $block, $strings, $target_lang );
+
+		$this->assertEquals( '<div>' . $translated_block_inner_HTML . '</div>', $updated_block->innerHTML );
```

Comment:
I think you can use `assertAttributeSame`/`assertAttributeSame` to  test the attribute of a DOM eLement.

Reply:
Tough, with the wrapper `<div>`, it might be problematic.

Reply:
I never used this kind of assertion, I'll probably use it in the future. Here, I prefer to keep testing the string because this is what we get as the output.